### PR TITLE
Support for S3 and Redshift

### PR DESCRIPTION
**Pull request by mkohn**

### CODE REVIEW

**Comment by mkohn** [comment]
Anything I should do to help get this merged?

File: leadbutt.py [hunk 1]
```diff
@@ -133,16 +134,31 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
...
-            end_time,
-            metric['MetricName'],  # RequestCount, CPUUtilization
-            metric['Namespace'],  # AWS/ELB, AWS/EC2
-            metric['Statistics'],  # Sum, Maximum
-            dimensions=metric['Dimensions'],
-            unit=metric['Unit'],  # Count, Percent
-        )
+
+        # if the metric is something that AWS only give you once per day
+        if metric['Namespace'] == "AWS/S3" or metric['Namespace'] == "AWS/Redshift":
```
Comment (paul-krohn) [inline leadbutt.py]: An array of 'once a day' metrics would be preferable; in `__init__.py` would be ideal, but we don't have that; in any case, loadable from the other script/module if possible?

File: sample_templates/ec2.yml.j2 [hunk 2]
```diff
@@ -14,6 +14,59 @@ Metrics:
     InstanceId: "{{ instance.id }}"
   Options:
     {#- I'm assuming my tag names are safe to use as metric names here #}
-    Formatter: 'cloudwatch.%(Namespace)s.{{ instance.tags['Name'] }}.%(MetricName)s.%(statistic)s.%(Unit)s'
+    Formatter: 'collectd.{{ instance.tags['Name'] }}.%(MetricName)s.%(statistic)s.%(Unit)s'
```
Comment (paul-krohn) [inline sample_templates/ec2.yml.j2]: If people are using these samples, the stat name change will take them by surprise. It also feels specific to your installation; is it possible to keep this consistent with the other templates?

File: sample_templates/ec2.yml.j2 [hunk 3]
```diff
@@ -14,6 +14,59 @@ Metrics:
...
     Period: 5
+
+- Namespace: "AWS/EC2"
+  MetricName: "DiskWriteOps"
```
Comment (paul-krohn) [inline sample_templates/ec2.yml.j2]: As an FYI, similar stats like DiskWriteOps and DiskReadOps are almost always safe to combine, eg:
```
  MetricName:
    - DiskWriteOps
    - DiskReadOps
```

Even stats with different units can be combined this way, leadbutt relies on the unit name returned from CloudWatch when it is not specified in the yaml/config.

File: sample_templates/ec2.yml.j2 (hunk 2 above)
Comment (mkohn) [inline sample_templates/ec2.yml.j2]: Yep your are correct. I thought i abstracted out all of these but I guess I missed it.

**Comment by paul-krohn** [comment]
Thanks for this! I had a branch with S3 support in it, but wasn't happy with it so never made a pull request.

Can you merge master in to your branch & resolve the conflicts? Then we'll have a good idea how it will look.

**Comment by mkohn** [comment]
That is great feed back and I appreciate it. I will work on these changes later this week when I get some time and resubmit.

**Pull request closed without merging**